File: src/fullstack/apps/documents/src/app/documents/models/document.ts
import { DocumentDto } from '../dto/document.dto';
import { DocumentType } from './document-type';

export class Document {
  id: string;
  number: string;
  createdAt: Date;
  author: string;
  type: DocumentType;

  constructor(dto: DocumentDto) {
    this.id = dto.id;
    this.number = dto.number;
    this.createdAt = new Date(dto.createdAt);
    this.author = dto.author;
    this.type = dto.type;
  }
}

File: src/fullstack/apps/documents-api/Features/Documents/Document.cs
namespace Fullstack.DocumentsApi.Features.Documents;

public class Document
{
  public Guid Id { get; }
  public string Number { get; private set; }
  public DateTimeOffset CreatedAt { get; private set; }
  public string Author { get; private set; }
  public DocumentType Type { get; private set; }

  public Document(string number, DateTimeOffset createdAt, string author, DocumentType type)
  {
    Id = Guid.NewGuid();
    Number = number;
    CreatedAt = createdAt;
    Author = author;
    Type = type;
  }
}
